framework/web/Controller.php

```diff
@@ -128,15 +128,37 @@ public function bindActionParams($action, $params)
         foreach ($method->getParameters() as $param) {
             $name = $param->getName();
             if (array_key_exists($name, $params)) {
+                $isValid = true;
                 if ($param->isArray()) {
-                    $args[] = $actionParams[$name] = (array) $params[$name];
-                } elseif (!is_array($params[$name])) {
-                    $args[] = $actionParams[$name] = $params[$name];
-                } else {
+                    $params[$name] = (array)$params[$name];
+                } elseif (is_array($params[$name])) {
+                    $isValid = false;
+                } elseif (PHP_VERSION_ID >= 70000 && ($type = $param->getType()) !== null) {
+                    if ($type->allowsNull() && ($params[$name] === null || $params[$name] === '')) {
```

Comment:
Are you sure about converting empty string to `null`? This is probably a BC break.

Reply:
Yeah you‘re right. Just pushed another commit that stops doing that. Also no longer doing anything at all with `bool` params, since those weren’t causing PHP errors in the first place. So now the only change in behavior this PR brings is that a `BadRequestHttpException` will be thrown in cases where a PHP error was going to occur otherwise.